### PR TITLE
Clear labels from admin panel, unarchive feature

### PR DESCRIPTION
**Pull request by iambibhas**

### CODE REVIEW

File: funnel/views/label.py [hunk 1]
```diff
@@ -172,6 +172,22 @@ def archive(self):
...
+        if not self.obj.archived:
+            flash(_("This label is not archived"), category='error')
+        else:
+            form = forms.Form()
+            if form.validate_on_submit():
+                self.obj.archived = False
+                db.session.commit()
+                flash(_("The label has been archived"), category='success')
+            else:
+                flash(_("CSRF token is missing"), category='error')
```
Comment (jace) [inline funnel/views/label.py]: Never tell a user "CSRF token is missing". What is the user supposed to do? There is no CSRF token in the UI. Instead, tell the user to submit again to confirm the action.

File: funnel/views/proposal.py [hunk 2]
```diff
@@ -62,9 +62,9 @@ def proposal_data(proposal):
             ('objective', proposal.objective.html),
             ('description', proposal.description.html),
             ('requirements', proposal.requirements.html),
-            ('slides', proposal.slides.url),
+            ('slides', proposal.slides.url if proposal.slides is not None else ''),
```
Comment (jace) [inline funnel/views/proposal.py]: This can be achieved with just `or ''`

File: funnel/views/proposal.py [hunk 3]
```diff
@@ -62,9 +62,9 @@ def proposal_data(proposal):
...
             ('links', proposal.links),
-            ('preview_video', proposal.preview_video.url),
+            ('preview_video', proposal.preview_video.url if proposal.preview_video is not None else ''),
```
Comment (jace) [inline funnel/views/proposal.py]: Same.

File: funnel/views/proposal.py (hunk 2 above)
Comment (iambibhas) [reply funnel/views/proposal.py]: That will break if `proposal.slides` is missing/`None`. which is why I added it.

File: funnel/forms/proposal.py [hunk 4]
```diff
@@ -10,6 +10,19 @@
     'ProposalMoveForm', 'ProposalLabelsAdminForm']
 
 
+class ClearableRadioField(forms.RadioField):
```
[the remark this answers is not in the document]
Comment (iambibhas) [reply funnel/forms/proposal.py]: It's not. I got to know about that decision later that fields can not be required in admin widget form. So this field is no longer needed. removing it.

File: funnel/views/proposal.py (hunk 2 above)
Comment (jace) [reply funnel/views/proposal.py]: That is what `or ''` solves for.

Comment (iambibhas) [reply funnel/views/proposal.py]: `proposal.slides.url` raises `AttributeError` is `proposal.slides` is not set.

Comment (jace) [reply funnel/views/proposal.py]: Ah, missed the `.url` bit.

**Pull request closed without merging**